clientv3/lease.go

```diff
@@ -119,6 +119,13 @@ func (l *lessor) Grant(ctx context.Context, ttl int64) (*LeaseGrantResponse, err
 	defer close(done)
 
 	for {
+		l.rc.mu.Lock()
+		closed := l.rc.client.cancel == nil
```

Comment:
this check isn't safe either-- if the client rlock isn't held for the span of checking if the client is closed and using the grpc LeaseClient then a Close can still happen in between and break it